### PR TITLE
Support source JIRA without Tempo, refactoring, Python 3 port

### PR DESCRIPTION
**Pull request by mmaslowskicc**

### CODE REVIEW

File: pylintrc [hunk 1]
```diff
@@ -1,2 +1,5 @@
 [FORMAT]
 max-line-length = 120
+
+[MESSAGES CONTROL]
+disable=R0205
```
Comment (fizyk) [inline pylintrc]:
```suggestion
disable=useless-object-inheritance
```

I prefer names, as they tend to tell what kind of errors we ignore.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
+FROM python:3.7 as app
+
+WORKDIR /usr/src/app
+# Install required dependencies and cache test dependencies, so they won't be redownloaded when updating after the code
+# is changed. (We cannot use setup.py here, since it requires other files from this package.)
+COPY requirements.txt requirements-tests.txt ./
+RUN pip install -r requirements.txt
+RUN pip download -r requirements-tests.txt
+COPY . .
+RUN pip install .
```
Comment (fizyk) [inline Dockerfile]: isn't `pip install .` running a setup.py which requires other files from this package?

File: requirements-tests.txt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+pytest==3.9.3
+pytest-cov==2.6.0
+pylint==1.9.3; python_version < "3"
```
Comment (fizyk) [inline requirements-tests.txt]: we can stop supporting python 2.7

**Comment by fizyk** [comment]
With dockerfile, and with tests actually in general we can set up CI

File: Dockerfile (hunk 2 above)
Comment (mmaslowskicc) [reply Dockerfile]: Yes, it's the reason for `COPY` above.

File: requirements-tests.txt (hunk 3 above)
Comment (mmaslowskicc) [reply requirements-tests.txt]: https://github.com/ClearcodeHQ/jira_timemachine/issues/11

**Pull request merged**